### PR TITLE
Add events API

### PR DESCRIPTION
**Pull request by guzzilar**
Add a basic events API

example

``` php
// Retrieve event list object
$events = OmiseEvent::retrieve();


// Retrieve event object by provided an event id
$event = OmiseEvent::retrieve('event_id');
```

### CODE REVIEW

**Comment by guzzilar** [comment]
@bank32 can you help me review this pr?

thanks

**Comment by bank32** [comment]
@guzzilar nothing to comment, you can merge it if you want. 

👍

File: tests/omise/EventTest.php [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+    $this->assertTrue(method_exists('OmiseEvent', 'reload'));
+    $this->assertTrue(method_exists('OmiseEvent', 'getUrl'));
+  }
+
+  /**
+   * Assert that a list of event object could be successfully retrieved.
+   *
+   */
+  public function testRetrieveEventListObject() {
+    $event = OmiseEvent::retrieve();
```
Comment (bank32) [inline tests/omise/EventTest.php]: @guzzilar how can you skip this line from authentication?, I have tried to do like this but I'm stuck in the authentication step.

Comment (nimid) [inline tests/omise/EventTest.php]: @bank32 For the authentication, is it mean it does real curl or connect to actual API?

Comment (bank32) [inline tests/omise/EventTest.php]: @nimid @guzzilar I got it, my bad. I ran phpunit from Ubuntu it doesn't work but when I run phpunit from composer, it works fine, just realized it's different.

**Comment by guzzilar** [comment]
about https://github.com/omise/omise-php/pull/31/commits/95099a2087c2d01c25f40e8947ccee15a013142d commit,
before fix
```php
/* Retrieve list */
$events = OmiseEvent::retrieve();
$events->reload();

echo $events['object']; // output: "list"

/* Retrieve item */
$events = OmiseEvent::retrieve('event_id');
$events->reload();

echo $events['object']; // output: "list"
```

after fixed
```php
/* Retrieve list */
$events = OmiseEvent::retrieve();
$events->reload();

echo $events['object']; // output: "list"

/* Retrieve item */
$events = OmiseEvent::retrieve('event_id');
$events->reload();

echo $events['object']; // output: "event" of "event_id"
```

**Comment by guzzilar** [comment]
for unit test, commit https://github.com/omise/omise-php/pull/31/commits/c314c0b0820d6eadaf2d28f23d79a0163c28a112

if run unit test without fix reload method
```shell
~/D/O/omise-php (feature/event-api) $ ./vendor/bin/phpunit tests/omise/EventTest.php 
PHPUnit 4.5.1 by Sebastian Bergmann and contributors.

Configuration read from /Omise/omise-php/phpunit.xml

....F

Time: 53 ms, Memory: 3.50MB

There was 1 failure:

1) EventTest::testReloadEventObjectGivenByEvenId
Failed asserting that two strings are equal.
--- Expected
+++ Actual
@@ @@
-'event'
+'list'

/Omise/omise-php/tests/omise/EventTest.php:58

FAILURES!
Tests: 5, Assertions: 12, Failures: 1.
```

after fixed with commit https://github.com/omise/omise-php/pull/31/commits/95099a2087c2d01c25f40e8947ccee15a013142d
result
```shell
~/D/O/omise-php (feature/event-api) $ ./vendor/bin/phpunit tests/omise/EventTest.php 
PHPUnit 4.5.1 by Sebastian Bergmann and contributors.

Configuration read from /Omise/omise-php/phpunit.xml

.....

Time: 56 ms, Memory: 3.25MB

OK (5 tests, 13 assertions)
```

**Comment by guzzilar** [comment]
Im going to merge it 👍 
If you guys has any comment/issue further, feel free to submit another PR to change.

thanks 😄

**Pull request merged**